tools/pkglint/lib/rules.ts

```diff
@@ -47,6 +47,37 @@ export class DescriptionIsRequired extends ValidationRule {
   }
 }
 
+/**
+ * Verify cdk.out directory is included in npmignore since we should not be
+ * publihsing it.
+ */
+export class CdkOutMustBeNpmIgnored extends ValidationRule {
+
+  public readonly name = 'package-info/npm-ignore-cdk-out';
+
+  public validate(pkg: PackageJson): void {
+
+    const npmIgnorePath = path.join(pkg.packageRoot, '.npmignore');
+
+    if (fs.existsSync(npmIgnorePath)) {
+
+      const npmIgnore = fs.readFileSync(npmIgnorePath);
+
+      if (!npmIgnore.includes('**/cdk.out')) {
+        pkg.report({
+          ruleName: this.name,
+          message: `${npmIgnorePath}: Must exclude **/cdk.out`,
+          fix: () => fs.writeFileSync(
+            npmIgnorePath,
+            `${npmIgnore}\n# exclude cdk artifacts\n**/cdk.out`
+          )
+        });
+      }
```

Comment:
No need to prefix with `**/`, and only ignore if it is a directory:

```suggestion
      if (!npmIgnore.includes('cdk.out/')) {
        pkg.report({
          ruleName: this.name,
          message: `${npmIgnorePath}: Must exclude cdk.out/`,
          fix: () => fs.writeFileSync(
            npmIgnorePath,
            `${npmIgnore}\n# exclude cdk artifacts\ncdk.out/`
          )
        });
      }
```

Reply:
@RomainMuller Are you sure about the `**/`? 

This is `npm pack` output without it:

```console
❯ npm pack                                                                                                  
npm notice 
npm notice 📦  @aws-cdk/aws-batch@0.0.0
npm notice === Tarball Contents === 
npm notice 193.3kB .jsii                               
npm notice 0       test/cdk.out/file                   
npm notice 0       test/nested/cdk.out/hello           
npm notice 11.4kB  LICENSE                             
npm notice 113B    NOTICE                              
....
....
aws-cdk-aws-batch-0.0.0.tgz
```

And this is with it:

```console
❯ npm pack                                                                                                  
npm notice 
npm notice 📦  @aws-cdk/aws-batch@0.0.0
npm notice === Tarball Contents === 
npm notice 193.3kB .jsii                               
npm notice 11.4kB  LICENSE                             
npm notice 113B    NOTICE                              
....
....
aws-cdk-aws-batch-0.0.0.tgz
```

Am I missing something?

Reply:
Oh fuck this is npmIgnore, not gitignore! Sorry this is PEBCAK.